lessons/codap-transformers2/langs/en-us/pages/codap-build.adoc

```diff
@@ -0,0 +1,84 @@
+= Build Attribute
+++++
+<style>
+body.workbookpage td .autonum::after { content: ')' !important; }
+/* Cram everything to the top instead of distributing space equally */
+#content { display: block !important; }
+</style>
+++++
+
+Make sure you’re logged into the @starter-file{animals} in CODAP. Select the Plugins icon, then choose Transformers.
+
+@vspace{2ex}
+
+== Create, Apply, and Save a Build Attribute Transformer (Step by Step)
+
+[cols="5, 10", frame="none", grid="none", stripes="none"]
+|===
+
+|
+@image{../images/build-ui-annotated.png, 350}
+
+|
+@n Choose Build Attribute from the drop-down menu (Box 1).
+
+@vspace{1ex}
+
+@n Enter `build-is-heavy` as the name (Box 2). What does this name tell you about the Transformer we are creating?
+
+@fitb{}{@ifsoln{It will build a column that states `true` for heavy animals and `false` for lightweight animals.}}
+
+@vspace{1ex}
+
+@n Select the dataset you'd like to transform (Box 3).
+
+@vspace{1ex}
+
+@n Let's name our new attribute `Heavy` (Box 4). What happened to the starred text (left) when you named the attribute?
+
+@fitb{}{}
+
+@vspace{1ex}
+
+@n Ensure that the collection you are adding to is "cases" (Box 5).
+
+@vspace{1ex}
+
+@n A domain is provided (row), but not a range. What is the desired output for `build-is-heavy`? @fitb{}{@ifsoln{Boolean.}}
+
+@vspace{1ex}
+
+@n Write a purpose statement (Box 7). What do we want the expression to do?
+
+@fitb{}{@ifsoln{Consume a weight and return `true` if the weight is greater than a specified amount.}}
+
+@vspace{1ex}
+
+@n Enter `Weight > 40` as the expression (Box 8).
+
+@vspace{1ex}
+
+@n Apply the Transformer. To define the Transformer for future use, select `Save`.
+
+|===
+
+@vspace{1ex}
+
+== More Building (On Your Own)
+
+@n Now, create a Transformer called `updated-age`. This Transformer will give the animals' ages one year from today. How many animals are 9 years-old one year from today?
+
+@fitb{}{@ifsoln{Three animals.}}
+
+@n Use a Transformer to build a column that indicates the number of letters in each animal's name. How many animals have exactly eight letters in their name?
+
```

Comment:
Do we care if they name it name-length or name length? Both are possible in CODAP. Worth mentioning best naming practice somewhere? perhaps @schanzer has language for this?